src/ipc/ipc-common/Registry.h

```diff
@@ -0,0 +1,113 @@
+/**********************************************************************
+**
+** Copyright (C) 2018 Luxoft Sweden AB
+**
+** This file is part of the FaceLift project
+**
+** Permission is hereby granted, free of charge, to any person
+** obtaining a copy of this software and associated documentation files
+** (the "Software"), to deal in the Software without restriction,
+** including without limitation the rights to use, copy, modify, merge,
+** publish, distribute, sublicense, and/or sell copies of the Software,
+** and to permit persons to whom the Software is furnished to do so,
+** subject to the following conditions:
+**
+** The above copyright notice and this permission notice shall be
+** included in all copies or substantial portions of the Software.
+**
+** THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+** EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+** MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+** NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+** BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+** ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+** CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+** SOFTWARE.
+**
+** SPDX-License-Identifier: MIT
+**
+**********************************************************************/
+#pragma once
+
+#include <QString>
+#include <QMap>
+#include <functional>
+#include <QVector>
+
+template<typename Type>
+class Registry {
```

Comment:
I do not want to rewrite/duplicate all the methods of QMap here. Only the methods which alter the content of the registry are defined here, whereas ALL const operations can be performed by calling the content() method.

Reply:
Wrapping (see also Adapter and Facade design patterns) is not duplicating methods. Moreover only operator[] and contains() are needed. In fact you just need to call them inside wrapping functions, the compiler will optimize it for you.

Reply:
Sure, but I would have to duplicate the method declarations, which is what I want to avoid. If I add operator[] (both const and non-const overloads) and contains(), one could argue that the other methods should also be added...

Reply:
None will argue, because you are creating your public interface for this class, QMap is only internal implementation (which should not be exposed in the future BTW - I believe there are going next steps in the future).
If the situation is different than that, then we don't need this class and we can just use QMap and a bunch of helper methods.